vcd/resource_vcd_vapp_firewall_rules_test.go

```diff
@@ -0,0 +1,473 @@
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.2.policy", "allow"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.2.protocol", "any"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.2.destination_port", "any"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.2.destination_vm_ip_type", "assigned"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.2.destination_vm_nic_id", "0"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.2.source_port", "any"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.2.source_vm_nic_id", "0"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.2.source_vm_ip_type", "NAT"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.2.enable_logging", "false"),
+
+					testAccCheckVcdVappFirewallRulesExists("vcd_vapp_firewall_rules."+t.Name(), params["Description4"].(string)),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.3.enabled", "false"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.3.policy", "drop"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.3.protocol", "any"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.3.destination_port", "any"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.3.destination_ip", "external"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.3.source_port", "any"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.3.source_ip", "internal"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name(), "rule.3.enable_logging", "true"),
+
+					testAccCheckVcdVappFirewallRulesExists("vcd_vapp_firewall_rules."+t.Name()+"2", params["Description1"].(string)),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name()+"2", "rule.0.enabled", "true"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name()+"2", "rule.0.policy", "drop"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name()+"2", "rule.0.protocol", "udp"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name()+"2", "rule.0.destination_port", "221"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name()+"2", "rule.0.destination_ip", "any"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name()+"2", "rule.0.source_port", "any"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name()+"2", "rule.0.source_ip", "10.10.0.10/24"),
+					resource.TestCheckResourceAttr("vcd_vapp_firewall_rules."+t.Name()+"2", "rule.0.enable_logging", "false"),
+				),
+			},
+			resource.TestStep{ // Step 2 - resource import
```

Comment:
Import can be done by ID or by name. We are only testing by name. We should test both.

Reply:
added

Reply:
added